File: fm-dddrive/src/main/java/io/dddrive/core/validation/model/impl/AggregatePartValidationImpl.java
package io.dddrive.core.validation.model.impl;

import org.springframework.lang.Nullable;

import io.dddrive.core.validation.model.AggregatePartValidation;
import io.dddrive.core.validation.model.enums.CodeValidationLevel;

public class AggregatePartValidationImpl implements AggregatePartValidation {

	private final Integer seqNr;
	private final CodeValidationLevel validationLevel;
	private final String message;
	private final String path;

	public AggregatePartValidationImpl(Integer seqNr, CodeValidationLevel validationLevel, String message) {
		this(seqNr, validationLevel, message, null);
	}

	public AggregatePartValidationImpl(Integer seqNr, CodeValidationLevel validationLevel, String message, @Nullable String path) {
		this.seqNr = seqNr;
		this.validationLevel = validationLevel;
		this.message = message;
		this.path = path;
	}

	@Override
	public Integer getSeqNr() {
		return this.seqNr;
	}

	@Override
	public CodeValidationLevel getValidationLevel() {
		return this.validationLevel;
	}


	@Override
	public String getMessage() {
		return this.message;
	}

	@Override
	@Nullable
	public String getPath() {
		return this.path;
	}

}

File: fm-dddrive/src/main/java/io/dddrive/core/validation/model/impl/AggregatePartValidationImpl.kt
package io.dddrive.core.validation.model.impl

import io.dddrive.core.validation.model.AggregatePartValidation
import io.dddrive.core.validation.model.enums.CodeValidationLevel

class AggregatePartValidationImpl(
	override val seqNr: Int,
	override val validationLevel: CodeValidationLevel,
	override val message: String,
	override val path: String? = null,
) : AggregatePartValidation
